theia-extensions/viewer-prototype/src/electron-node/trace-server/electron-trace-server-service.ts

```diff
@@ -0,0 +1,15 @@
+import { injectable } from '@theia/core/shared/inversify';
+import { dirname, resolve } from 'path';
+import { TraceServerServiceImpl } from '../../node/trace-server/trace-server-service';
+
+const APPLICATION_ROOT = resolve(dirname(process.argv0));
+
+export const BUNDLED_TRACE_SERVER_PATH = resolve(APPLICATION_ROOT, 'resources/trace-compass-server/tracecompass-server');
+
+@injectable()
+export class ElectronTraceServerServiceImpl extends TraceServerServiceImpl {
+
+    protected getTraceServerPath(): string | undefined {
+        return super.getTraceServerPath() || BUNDLED_TRACE_SERVER_PATH;
+    }
```

Comment:
Functionally, this is fine. To accommodate applications that don't bundle the server, this function could be retyped to return a `Promise` and check whether anything exists at `BUNDLED_TRACE_SERVER_PATH` and return `undefined` if not. That way, the 'No path provided' message would show rather than the error thrown by `spawn` with a bad path, since the latter message might leave users wondering where the strange temporary directory path came from.

Reply:
Made it a promise and added the disk check.